engine/services/thread_service.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "common/repository/thread_repository.h"
+#include "common/thread_tool_resources.h"
+#include "common/variant_map.h"
+#include "utils/result.hpp"
+
+class ThreadService {
+ public:
+  explicit ThreadService(std::shared_ptr<ThreadRepository> thread_repository)
+      : thread_repository_{thread_repository} {}
+
+  cpp::result<OpenAi::Thread, std::string> CreateThread(
+      std::optional<std::unique_ptr<OpenAi::ThreadToolResources>>
```

Comment:
Why do we need optional? I think can pass std::nullptr if we use unique_ptr?

Reply:
ah yeah, my mistake